libdevcore/CMakeLists.txt

```diff
@@ -2,7 +2,9 @@ file(GLOB sources "*.cpp")
 file(GLOB headers "*.h")
 
 add_library(devcore ${sources} ${headers})
-target_link_libraries(devcore PRIVATE jsoncpp ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(devcore PRIVATE ${JSONCPP_LIBRARY} ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 target_include_directories(devcore PUBLIC "${CMAKE_SOURCE_DIR}")
+target_include_directories(devcore PUBLIC "${JSONCPP_INCLUDE_DIR}")
```

Comment:
I'm not how this works, but basically `libdevcore/JSON.h` and `json/json.h` are included in many different libraries (libevmasm, libsolidity, libsolc, test), so I'd think the include directory should be made valid for all of them, e.g. should be put into the main CMake file and not into libdevcore.

The same applies to jsoncpp library then.

How to accomplish this?

Reply:
Add `inlcude_directories(${JSONCPP_INCLUDE_DIR})` just after `include(ProjectJsoncpp)`.

Reply:
No. Only iclude path.
You still have to use `target_link_libraries()`. But should be enough to specify that in devcore, because for static libs this dependency in transitive.

Reply:
I cannot find a `include(ProjectJsoncpp)` line? There is only a `include(jsoncpp)` line in the main cmake file.

Reply:
That's what I meant.